### PR TITLE
Add support for client keys #274

### PR DESCRIPTION
**Pull request by Goncharo**
Ticket: metriport/metriport-internal#274

### Dependencies

- Upstream: N/A
- Downstream: https://github.com/metriport/metriport-internal/pull/278

### Description

Add support for client keys for mobile clients that will be sending data to the Metriport API.

Updating docs is still a TODO here.

### Release Plan

- [ ] Merge this PR.
- [ ] Merge downstream deps.

### CODE REVIEW

**Review by Orta21 — APPROVED** [review]

File: infra/lib/api-stack.ts [hunk 1]
```diff
@@ -375,6 +375,47 @@ export class APIStack extends Stack {
...
+      integrationHttpMethod: "POST",
+      uri: `http://${fargateService.loadBalancer.loadBalancerDnsName}${appleHealthResource.path}`,
+    });
+    appleHealthResource.addMethod("POST", integrationApple, {
+      apiKeyRequired: true,
+    });
+
+    // add another usage plan for Publishable (Client) API keys
+    // everything is throttled to 0 - except explicitely permitted routes
+    const appleHealthThrottleKey = `${appleHealthResource.path}/POST`;
```
Comment (leite08) [inline infra/lib/api-stack.ts:395]: Can we move these to a function? This main function is super large and is getting hard to understand/maintain.

File: infra/lib/api-stack.ts [hunk 2]
```diff
@@ -375,6 +375,47 @@ export class APIStack extends Stack {
...
+      ],
+      throttle: {
+        burstLimit: 10,
+        rateLimit: 50,
+      },
+      quota: {
+        limit: this.isProd(props) ? 10000 : 500,
+        period: apig.Period.DAY,
+      },
+    });
```
Comment (leite08) [inline infra/lib/api-stack.ts:417]: This one could also go to its own function, receiving an array of `throttleKey` to have limits applied to.

**Review by leite08 — CHANGES_REQUESTED** [review]
Overall looks good, just would be nice to separate the new code on their own functions.

File: infra/lib/api-stack.ts (hunk 1 above)
Comment (Goncharo) [reply infra/lib/api-stack.ts:395]: For sure, will update

File: infra/lib/api-stack.ts (hunk 2 above)
Comment (Goncharo) [reply infra/lib/api-stack.ts:417]: Good idea, will update

File: infra/lib/api-stack.ts (hunk 1 above)
Comment (Goncharo) [reply infra/lib/api-stack.ts:395]: Will address this after merge to unblock staging deployments

File: infra/lib/api-stack.ts (hunk 2 above)
Comment (Goncharo) [reply infra/lib/api-stack.ts:417]: Will address this after merge to unblock staging deployments

**Pull request merged**